### PR TITLE
Add FBE baseline policy for geometric exploration

### PR DESCRIPTION
**Pull request by srama2512**
## Motivation and Context

This PR adds a baseline policy for the task of area-coverage exploration.

## Changelog

This adapts the semantic exploration policy by removing the segmentation model and semantic category channels in the map. 

## How Has This Been Tested

The policy has been tested on a Stretch 2 robot locally. 

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] I ran `tests/hw_manual_test.py` on hardware and observed that robot behavior looks normal.

### CODE REVIEW

File: src/home_robot/home_robot/mapping/geometric/categorical_2d_geometric_map_state.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import numpy as np
+import torch
+
+from home_robot.mapping.map_utils import MapSizeParameters, init_map_and_pose_for_env
+from home_robot.mapping.semantic.constants import MapConstants as MC
+
+
+class Categorical2DGeometricMapState:
```
Comment (cpaxton) [inline src/home_robot/home_robot/mapping/geometric/categorical_2d_geometric_map_state.py:12]: This is no longer categorical right? should it be renamed?

Comment (cpaxton) [reply src/home_robot/home_robot/mapping/geometric/categorical_2d_geometric_map_state.py:12]: geometric_map_state seems ok

File: src/home_robot_hw/home_robot_hw/env/visualizer.py [hunk 2]
```diff
@@ -278,3 +278,211 @@ def _init_vis_image(self, goal_name: str):
         # vis_image[537 : 537 + lx, 155 : 155 + ly, :] = self.legend
 
         return vis_image
+
+
+class ExplorationVisualizer:
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/env/visualizer.py:283]: I would really like to combine this with the other one somehow so we dont have so much duplicated code...

@theophilegervet  @yvsriram  we talked about deleting these and combining with the habitat one right? any thoughts?

File: src/home_robot_hw/home_robot_hw/env/stretch_exploration_env.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        # GPS in robot coordinates
+        gps = relative_pose.translation()[:2]
+
+        # Create the observation
+        obs = home_robot.core.interfaces.Observations(
+            rgb=rgb.copy(),
+            depth=depth.copy(),
+            gps=gps,
+            compass=np.array([theta]),
+            # base_pose=sophus2obs(relative_pose),
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/env/stretch_exploration_env.py:104]: should we clean up stuff like this that's commented out from the previous env?

File: projects/stretch_exploration/configs/agent/floorplanner_eval.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+BASE_TASK_CONFIG_PATH: configs/task/floorplanner_val.yaml
```
Comment (cpaxton) [inline projects/stretch_exploration/configs/agent/floorplanner_eval.yaml:1]: this is not for you but we really should rename these config files

**Review by cpaxton — COMMENTED** [review]
some minor nits and discussion points

File: src/home_robot_hw/home_robot_hw/env/stretch_exploration_env.py (hunk 3 above)
Comment (srama2512) [reply src/home_robot_hw/home_robot_hw/env/stretch_exploration_env.py:108]: This is actually commented out even in the previous env :)

File: src/home_robot_hw/home_robot_hw/env/stretch_exploration_env.py [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+        return obs
+
+    @property
+    def episode_over(self) -> bool:
+        pass
+
+    def get_episode_metrics(self) -> Dict:
+        pass
+
+    def rotate(self, theta):
```
[the remark this answers is not in the document]
Comment (srama2512) [reply src/home_robot_hw/home_robot_hw/env/stretch_exploration_env.py:121]: It looks like every task environment has get_episode_metrics and rotate methods. Is there a specific reason to remove from this particular env? Or do we remove them from all envs if they're not useful?

**Review by cpaxton — APPROVED** [review]
we should add some unit tests for geometric state + map module, and exploration agent - but I'm ok with going that after we've had time to discuss

**Review by cpaxton — APPROVED** [review]
lgtm, although we need a plan for adding tests

File: src/home_robot_hw/home_robot_hw/env/stretch_exploration_env.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (cpaxton) [reply src/home_robot_hw/home_robot_hw/env/stretch_exploration_env.py:120]: No i think they need to be implemented

**Pull request merged**